src/main.cpp

```diff
@@ -4513,9 +4535,18 @@ void wait_at_frontend(void)
     if ((game.operation_flags & GOF_SingleLevel) != 0)
     {
       faststartup_network_game();
+      if (restart_level)
```

Comment:
This is looking really weird. Could you explain?

Reply:
```
if (restart_level) {
        restart_level = false;
        }
```
Do both of these need to be restart_level? Why would you even need the if statement in this case, you could just set restart_level to false if not. I'm thinking you want one to reference a function and the other to a variable, and if that's the case, you want to have them be different names.

Reply:
One of the if (restart_level)s is in another if statement and before a return statement (though I suppose I could move this to before that if statement).

Which one should reference a function? restart_level is a boolean variable.